File: escape_sequences/go/control_sequences.go
// License: https://unlicense.org/
// Source:  https://github.com/jackiboi307/escape-sequences

package main

// Control codes

const BELL string = "\x07"

// Cursor control

const CUR_HOME string = "\x1b[H"
const CUR_SET string = "\x1b[%s;%sH"
const CUR_UP string = "\x1b[%sA"
const CUR_DOWN string = "\x1b[%sB"
const CUR_RIGHT string = "\x1b[%sC"
const CUR_LEFT string = "\x1b[%sD"
const CUR_DOWN_BEG string = "\x1b[%sE"
const CUR_UP_BEG string = "\x1b[%sF"
const CUR_COL string = "\x1b[%sG"

const CUR_UP_ONE string = "\x1b[1A"
const CUR_DOWN_ONE string = "\x1b[1B"
const CUR_RIGHT_ONE string = "\x1b[1C"
const CUR_LEFT_ONE string = "\x1b[1D"

const CUR_COL_HOME string = "\x1b[0G"

const CUR_HIDE string = "\x1b[?25l"
const CUR_SHOW string = "\x1b[?25h"

const CUR_SAVE string = "\x1b7"
const CUR_RESTORE string = "\x1b8"
const CUR_SAVE_SCO string = "\x1b[s"
const CUR_RESTORE_SCO string = "\x1b[u"

// Screen operations

const SCREEN_SAVE string = "\x1b[?47h"
const RESTORE_SAVE string = "\x1b[?47l"

const ERASE_SCREEN string = "\x1b[2J"
const ERASE_LINE string = "\x1b[2K"
const ERASE_TO_END string = "\x1b[0J"
const ERASE_TO_LINE_END string = "\x1b[0J"

// Style

const RESET string = "\x1b[0m"
const BOLD string = "\x1b[1m"
const DIM string = "\x1b[2m"
const ITALIC string = "\x1b[3m"
const UNDERLINE string = "\x1b[4m"
const BLINK string = "\x1b[5m"
const STRIKE string = "\x1b[9m"

// 256 colors

const FG_ID string = "\x1b[38;5;%sm"
const BG_ID string = "\x1b[48;5;%sm"

// True color

const FG_RGB string = "\x1b[38;2;%s;%s;%sm"
const BG_RGB string = "\x1b[48;2;%s;%s;%sm"

// 16 color codes

const FG_BLACK string = "\x1b[30m"
const FG_RED string = "\x1b[31m"
const FG_GREEN string = "\x1b[32m"
const FG_YELLOW string = "\x1b[33m"
const FG_BLUE string = "\x1b[34m"
const FG_MAGENTA string = "\x1b[35m"
const FG_CYAN string = "\x1b[36m"
const FG_WHITE string = "\x1b[37m"
const FG_DEFAULT string = "\x1b[39m"

const BG_BLACK string = "\x1b[40m"
const BG_RED string = "\x1b[41m"
const BG_GREEN string = "\x1b[42m"
const BG_YELLOW string = "\x1b[43m"
const BG_BLUE string = "\x1b[44m"
const BG_MAGENTA string = "\x1b[45m"
const BG_CYAN string = "\x1b[46m"
const BG_WHITE string = "\x1b[47m"
const BG_DEFAULT string = "\x1b[49m"

// Bright versions

const FG_BLACK_B string = "\x1b[90m"
const FG_RED_B string = "\x1b[91m"
const FG_GREEN_B string = "\x1b[92m"
const FG_YELLOW_B string = "\x1b[93m"
const FG_BLUE_B string = "\x1b[94m"
const FG_MAGENTA_B string = "\x1b[95m"
const FG_CYAN_B string = "\x1b[96m"
const FG_WHITE_B string = "\x1b[97m"

const BG_BLACK_B string = "\x1b[100m"
const BG_RED_B string = "\x1b[101m"
const BG_GREEN_B string = "\x1b[102m"
const BG_YELLOW_B string = "\x1b[103m"
const BG_BLUE_B string = "\x1b[104m"
const BG_MAGENTA_B string = "\x1b[105m"
const BG_CYAN_B string = "\x1b[106m"
const BG_WHITE_B string = "\x1b[107m"

File: escape_sequences/rust/control_sequences.rs
// License: https://unlicense.org/
// Source:  https://github.com/jackiboi307/escape-sequences

#![allow(dead_code)] // disable warnings for unused code

// Control codes

pub const BELL: &'static str = "\x07";

// Cursor control

pub const CUR_HOME: &'static str = "\x1b[H";
pub const CUR_SET: &'static str = "\x1b[{};{}H";
pub const CUR_UP: &'static str = "\x1b[{}A";
pub const CUR_DOWN: &'static str = "\x1b[{}B";
pub const CUR_RIGHT: &'static str = "\x1b[{}C";
pub const CUR_LEFT: &'static str = "\x1b[{}D";
pub const CUR_DOWN_BEG: &'static str = "\x1b[{}E";
pub const CUR_UP_BEG: &'static str = "\x1b[{}F";
pub const CUR_COL: &'static str = "\x1b[{}G";

pub const CUR_UP_ONE: &'static str = "\x1b[1A";
pub const CUR_DOWN_ONE: &'static str = "\x1b[1B";
pub const CUR_RIGHT_ONE: &'static str = "\x1b[1C";
pub const CUR_LEFT_ONE: &'static str = "\x1b[1D";

pub const CUR_COL_HOME: &'static str = "\x1b[0G";

pub const CUR_HIDE: &'static str = "\x1b[?25l";
pub const CUR_SHOW: &'static str = "\x1b[?25h";

pub const CUR_SAVE: &'static str = "\x1b7";
pub const CUR_RESTORE: &'static str = "\x1b8";
pub const CUR_SAVE_SCO: &'static str = "\x1b[s";
pub const CUR_RESTORE_SCO: &'static str = "\x1b[u";

// Screen operations

pub const SCREEN_SAVE: &'static str = "\x1b[?47h";
pub const RESTORE_SAVE: &'static str = "\x1b[?47l";

pub const ERASE_SCREEN: &'static str = "\x1b[2J";
pub const ERASE_LINE: &'static str = "\x1b[2K";
pub const ERASE_TO_END: &'static str = "\x1b[0J";
pub const ERASE_TO_LINE_END: &'static str = "\x1b[0J";

// Style

pub const RESET: &'static str = "\x1b[0m";
pub const BOLD: &'static str = "\x1b[1m";
pub const DIM: &'static str = "\x1b[2m";
pub const ITALIC: &'static str = "\x1b[3m";
pub const UNDERLINE: &'static str = "\x1b[4m";
pub const BLINK: &'static str = "\x1b[5m";
pub const STRIKE: &'static str = "\x1b[9m";

// 256 colors

pub const FG_ID: &'static str = "\x1b[38;5;{}m";
pub const BG_ID: &'static str = "\x1b[48;5;{}m";

// True color

pub const FG_RGB: &'static str = "\x1b[38;2;{};{};{}m";
pub const BG_RGB: &'static str = "\x1b[48;2;{};{};{}m";

// 16 color codes

pub const FG_BLACK: &'static str = "\x1b[30m";
pub const FG_RED: &'static str = "\x1b[31m";
pub const FG_GREEN: &'static str = "\x1b[32m";
pub const FG_YELLOW: &'static str = "\x1b[33m";
pub const FG_BLUE: &'static str = "\x1b[34m";
pub const FG_MAGENTA: &'static str = "\x1b[35m";
pub const FG_CYAN: &'static str = "\x1b[36m";
pub const FG_WHITE: &'static str = "\x1b[37m";
pub const FG_DEFAULT: &'static str = "\x1b[39m";

pub const BG_BLACK: &'static str = "\x1b[40m";
pub const BG_RED: &'static str = "\x1b[41m";
pub const BG_GREEN: &'static str = "\x1b[42m";
pub const BG_YELLOW: &'static str = "\x1b[43m";
pub const BG_BLUE: &'static str = "\x1b[44m";
pub const BG_MAGENTA: &'static str = "\x1b[45m";
pub const BG_CYAN: &'static str = "\x1b[46m";
pub const BG_WHITE: &'static str = "\x1b[47m";
pub const BG_DEFAULT: &'static str = "\x1b[49m";

// Bright versions

pub const FG_BLACK_B: &'static str = "\x1b[90m";
pub const FG_RED_B: &'static str = "\x1b[91m";
pub const FG_GREEN_B: &'static str = "\x1b[92m";
pub const FG_YELLOW_B: &'static str = "\x1b[93m";
pub const FG_BLUE_B: &'static str = "\x1b[94m";
pub const FG_MAGENTA_B: &'static str = "\x1b[95m";
pub const FG_CYAN_B: &'static str = "\x1b[96m";
pub const FG_WHITE_B: &'static str = "\x1b[97m";

pub const BG_BLACK_B: &'static str = "\x1b[100m";
pub const BG_RED_B: &'static str = "\x1b[101m";
pub const BG_GREEN_B: &'static str = "\x1b[102m";
pub const BG_YELLOW_B: &'static str = "\x1b[103m";
pub const BG_BLUE_B: &'static str = "\x1b[104m";
pub const BG_MAGENTA_B: &'static str = "\x1b[105m";
pub const BG_CYAN_B: &'static str = "\x1b[106m";
pub const BG_WHITE_B: &'static str = "\x1b[107m";
